src/query/__tests__/RelayQueryRoot-test.js

```diff
@@ -73,6 +73,38 @@ describe('RelayQueryRoot', () => {
     expect(children[2].isGenerated()).toBe(true);
   });
 
+  it('does not return skipped children', () => {
+    var query = getNode(Relay.QL`
+      query {
+        me {
+          id,
+          firstName @skip(if: $cond),
+          lastName @include(if: $cond2),
```

Comment:
nit: variable names are arbitrary, so you could name this `@skip(if: $true)` to make it even more clear that the variable's value will be `true`